source/beamline/CLS/CLSShutters.cpp

```diff
@@ -1,11 +1,16 @@
 #include "CLSShutters.h"
 
 #include "beamline/AMControl.h"
+#include "actions3/AMActionSupport.h"
 #include "actions3/AMListAction3.h"
 
+#include "util/AMErrorMonitor.h"
+
 CLSShutters::CLSShutters(const QString &name, QObject *parent) :
 	CLSTriStateGroup(name, parent)
 {
+	safetyShutter_ = 0;
```

Comment:
@dretrex my understanding for this is that it is up to the user to make the decision how to use the safety shutter for CLSShutters.
For SXRMB:
The safety shutter is a Readonly PV control, which can't be turned on grammatically. The users complained several times that they can't turn beam on with acquaman. Acutally, the safety shutter is closed, but they don't know. With this changes, we will visually display the status of the safety shutter and also pop a message to tell them if the safety shutter is closed when they are trying to turn beam on.

For VESPERS, since we are able to open the safety shutter, we don't have to set the safety shutter, which can be identified as a regular shutter. If we still want to distinguish it with the other shutters, it will still works as I mentioned before.

To address @helfrij's question, safety shutter is not mandantory to the `CLSShutters`. You can use `CLSShutters` as what you did with the original `BioXASShutters`. It can be simply a group of shutters you want.